include/boost/numeric/ublas/tensor/fixed_rank_tensor.hpp

```diff
@@ -0,0 +1,261 @@
+
+        using tensor_temporary_type     = self_type;
+        using storage_category          = dense_tag;
+
+        using extents_type              = typename detail::tensor_traits<self_type>::extents_type;
+        using strides_type              = strides_t<extents_type,layout_type>;
+
+        using matrix_type               = typename super_type::matrix_type;
+        using vector_type               = typename super_type::vector_type;
+
+        constexpr fixed_rank_tensor() = default;
+
+        constexpr fixed_rank_tensor( extents_type const& e, value_type const& i )
+            : super_type(e,i)
+        {}
+
+        constexpr fixed_rank_tensor( extents_type const& e, array_type const& a )
+            : super_type(e,a)
+        {}
+
+        constexpr fixed_rank_tensor( extents_type const& e )
+            : super_type(e)
+        {}
+
+        fixed_rank_tensor( std::initializer_list< size_type > li )
+            : super_type( std::move(li) )
+        {}
+
+        constexpr fixed_rank_tensor( matrix_type const& v )
+            : super_type( extents_type{v.size1(), v.size2()} )
+        {
+            static_assert( R == 2, "Error in boost::numeric::ublas::tensor(const matrix &v)"
+                                            " : rank of extents are not correct, please check!");
+
+            std::copy(v.data().begin(), v.data().end(), super_type::begin());
+        }
+
+        constexpr fixed_rank_tensor( matrix_type && v )
+            : super_type( extents_type{v.size1(), v.size2()} )
+        {
+            
+            static_assert( R == 2, "Error in boost::numeric::ublas::tensor(matrix &&v)"
+                                            " : rank of extents are not correct, please check!");
+
+            auto const sz = v.size1() * v.size2();
+            for(auto i = size_type{}; i < sz; ++i){
+                super_type::data_[i] = std::move(v.data()[i]);
+            }
+        }
+
+        constexpr fixed_rank_tensor (const vector_type &v)
+            : super_type( extents_type{ v.size(), typename extents_type::value_type{1} } )
+        {
+            static_assert( R == 2, "Error in boost::numeric::ublas::tensor(const vector &v)"
+                                            " : rank of extents are not correct, please check!");
+
+            std::copy(v.data().begin(), v.data().end(), super_type::begin());
+            
+        }
+
+        constexpr fixed_rank_tensor (vector_type &&v)
+            : super_type( extents_type{ v.size(), typename extents_type::value_type{1} } )
+        {
+            
+            static_assert( R == 2, "Error in boost::numeric::ublas::tensor(vector &&v)"
+                                            " : rank of extents are not correct, please check!");
+
+            auto const sz = v.size();
+            for(auto i = size_type{}; i < sz; ++i){
+                super_type::data_[i] = std::move(v.data()[i]);
+            }
```

Comment:
1. I recommend to use the `std::move` algorithm
2. I recommend to immediately initialize the `super_type` with the data in the initialization list `:`.

Reply:
1. Ok
2. Same as above.